ts/features/bonus/bpd/screens/details/transaction/BpdTransactionsScreen.tsx

```diff
@@ -1,33 +1,164 @@
 import * as pot from "italia-ts-commons/lib/pot";
 import { View } from "native-base";
 import * as React from "react";
-import { FlatList, SafeAreaView } from "react-native";
+import {
+  SafeAreaView,
+  SectionList,
+  SectionListData,
+  SectionListRenderItem
+} from "react-native";
 import { connect } from "react-redux";
 import { Dispatch } from "redux";
 import { compareDesc } from "date-fns";
-import { index } from "fp-ts/lib/Array";
+import { index, reverse } from "fp-ts/lib/Array";
+import { fromNullable } from "fp-ts/lib/Option";
 import { H1 } from "../../../../../../components/core/typography/H1";
 import { IOStyles } from "../../../../../../components/core/variables/IOStyles";
 import BaseScreenComponent from "../../../../../../components/screens/BaseScreenComponent";
 import I18n from "../../../../../../i18n";
 import { GlobalState } from "../../../../../../store/reducers/types";
 import BPDTransactionSummaryComponent from "../../../components/BPDTransactionSummaryComponent";
+import { format } from "../../../../../../utils/dates";
+import BaseDailyTransactionHeader from "../../../components/BaseDailyTransactionHeader";
 import {
   BpdTransactionItem,
   EnhancedBpdTransaction
 } from "../../../components/transactionItem/BpdTransactionItem";
 import { bpdAmountForSelectedPeriod } from "../../../store/reducers/details/amounts";
 import { bpdDisplayTransactionsSelector } from "../../../store/reducers/details/combiner";
 import { bpdSelectedPeriodSelector } from "../../../store/reducers/details/selectedPeriod";
-import { format } from "../../../../../../utils/dates";
+import { IOColors } from "../../../../../../components/core/variables/IOColors";
 
 export type Props = ReturnType<typeof mapDispatchToProps> &
   ReturnType<typeof mapStateToProps>;
 
+type TotalCashbackPerDate = {
+  trxDate: Date;
+  totalCashBack: number;
+};
+
 const dataForFlatList = (
   transactions: pot.Pot<ReadonlyArray<EnhancedBpdTransaction>, Error>
 ) => pot.getOrElse(transactions, []);
 
+export const isTotalCashback = (item: any): item is TotalCashbackPerDate =>
+  item.totalCashBack;
```

Comment:
Since it should return a boolean don't we have to check if `totalCashBack` should be defined?
```suggestion
  item.totalCashBack !== undefined;
```